src/lit-html.ts

```diff
@@ -307,11 +307,7 @@ export class NodePart extends Part {
             // Since this is the last part we'll use, set it's endNode to the
             // container's endNode. Setting the value of this part will clean
             // up any residual nodes from a previously longer iterable.
-            const range = document.createRange();
-            range.setStartBefore(itemPart.endNode);
-            range.setEndBefore(this.endNode);
-            range.deleteContents();
-            range.detach();
+            this.clear(itemPart.endNode);
```

Comment:
It'd be nice to have a test that failed for this. I have a few where I check the childNodes of the container, not just innerHTML, so we're sure there's no orphan marker nodes hanging around